### DISCUSSION TITLE
Intelliflo Pump Settings?

### DISCUSSION THREAD

**Discussion by dcorrea78** [tagyoureit/nodejs-poolController#755] [App and O/S Config Questions]
I have nodejs-poolController and the nodejs-poolController-dashPanel installed and running successfully on a RPi4. My system has a *touch 2-8 to control a spa and pool, and my pump is a Intelliflo VSF pump. When using the dashboard, the pump settings show the wrong circuits, and I cannot manually change circuit or change the pump speed. I have tried to add the correct pool circuit, but whenever I update the pump it just spins and never updates. Is there a way to manually set the pump speed and update the pump circuits? What am I doing wrong?
![Screenshot 2023-05-07 5 45 57 PM](https://user-images.githubusercontent.com/34433189/236704112-244ee5f2-9758-46d1-a5fe-af403304aa52.png)
![Screenshot 2023-05-07 5 46 37 PM](https://user-images.githubusercontent.com/34433189/236704115-73ad1330-7cdb-4edd-9448-b9a910762bff.png)

**Comment by tagyoureit** [comment]
This should work.  There could be some corruption in your data files.  Can you upload a [replay](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) here?  

Also, if you enable the javascript console you may see an error in the logs.

**Comment by dcorrea78** [comment]
I got it fixed. I reloaded the config and now it works properly. Thanks

**Comment by dcorrea78** [comment]
I spoke too soon. I was working on some rule machine rules for my hubitat dashboard today when the nodejs-dash reverted back to the same behavior. What do you recommend? Any chance there's logs showing what happened when it auto reverted? If so, where would they be?

**Comment by rstrouse** [comment]
> Any chance there's logs showing what happened when it auto reverted? If so, where would they be?

There is not but perhaps we may see something from a replay.  Specifically how the data is corrupted.

**Reply by dcorrea78** [reply]
[replay (3).zip](https://github.com/tagyoureit/nodejs-poolController/files/11436499/replay.3.zip)
Replay attached.

**Reply by rstrouse** [reply]
I am not seeing a corrupted configuration like you have above.  The pump has Pool, Spa, and High Speed circuits.  I do see SSL connection errors to Hubitat in the log though.

**Reply by dcorrea78** [reply]
Interesting. I'll take a look. I had to stop the nodejs apps because my pool control was going crazy. I'll start it up again and see if the issue is on the hubitat connection.

**Comment by dcorrea78** [comment]
Is there an easy way to nuke everything and start over? I have everything running at startup with pm2, so after running 'pm2 unstartup' and pasting the output, do i just uninstall/reinstall npm to start over?

**Reply by tagyoureit** [reply]
You can wipe out any/all of the nodejs-poolController, nodejs-poolController-dashPanel, and (if you were using it relayEquipmentManager) directories.  Then just reinstall from scratch.  And for clarity, NPM is typically installed as part of the NodeJS core package.  Unless you are changing versions/upgrading Node you don't need to reinstall NPM.

[PM2](https://github.com/tagyoureit/nodejs-poolController/wiki/Automatically-start-at-boo---PM2-&-Systemd#pm2) you also don't likely need to reinstall.  You can just set it up to run the applications after you have them installed.

**Reply by dcorrea78** [reply]
Can you take a look at this replay? I deleted the nodejs-poolcontroller and nodejs-poolcontroller-dashPanel directories, re-cloned the git repos for both, and then re-installed both packages. 

After about 10-15 minutes of normal activity, the poolcontoller reverted to the same behavior as before, except instead of showing light, spa and high speed, it now showed spa, spa, and high speed. 

After that, I disconnected the Hubitat integration and deleted the pool controller app from Hubitat to make sure this isn't caused by the Hubitat integration somehow. 

After rebooting the RPi, everything started normally. I started the replay and took my dog for a walk. When I got back the poolcontroller reverted to the behavior. 

To note, the pool was off during this time except for the lights. 
The first screenshot shows the incorrect behavior, and the second one shows the proper config after rebooting the RPi. 

![Screenshot 2023-05-10 10 48 17 PM](https://github.com/tagyoureit/nodejs-poolController/assets/34433189/f006fc69-8b18-46de-97d7-0b60f16fa0a3)
[replay (4).zip](https://github.com/tagyoureit/nodejs-poolController/files/11447677/replay.4.zip)
![image](https://github.com/tagyoureit/nodejs-poolController/assets/34433189/2be2eaa7-7379-44a8-ad21-d35ec204569d)

**Comment by johnny2678** [comment]
I've also noticed the errant 31,XXX RPM # assigned to circuits before but hasn't happened in a week or so.  

Even now, my rpm settings page is showing the incorrect rpms for the SPA (supposed to be 2860, not 450).  It does actually run at 2860, so that's good.

And my lights don't have an rpm assigned to the circuit 🤷‍♂️:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/729e1cd3-1ec1-413b-87db-c1b87b1602eb)

RPM settings tab from screen logic:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/0079d542-1e86-4d0c-85a2-04ace11e51bb)

Happy to provide a replay, just let me know what you want me to capture.

Oh, and I'm not running the hubitat integration like @dcorrea78 but I do have an easytouch-4

**Reply by rstrouse** [reply]
Yes please give me a replay.  This has to be some strange byte alignment,

**Reply by rstrouse** [reply]
I need a replay when this is not correct.  It would be great if you catch it when it is not correct in njsPC and change it in ScreenLogic.  All of the previous replays have been when there was not a problem.

**Comment by dcorrea78** [comment]
So my issue has got worse. So after the pump started showing the SPA circuit twice, I could no longer change pump speeds and got an InvalidEquipmentData errror. So I decided to just nuke the whole thing and start over with a clean install.

Unfortunately it is doing the same thing. I can go into the settings and change the circuit back to the pool circuit, but it changes back after a few minutes. So at this point, changing the pump RPM is broken for me. 

> ApiError: Configuration for pump Pump1 is not correct circuit #1 as included more than once. {"speed":400,"units":0,"id":2,"circuit":1}
    at TouchPumpCommands.setPumpAsync (/home/dnm/nodejs-poolController/controller/boards/EasyTouchBoard.ts:2464:88)
    at /home/dnm/nodejs-poolController/web/services/config/Config.ts:573:50
    at Layer.handle [as handle_request] (/home/dnm/nodejs-poolController/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/dnm/nodejs-poolController/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/home/dnm/nodejs-poolController/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/home/dnm/nodejs-poolController/node_modules/express/lib/router/layer.js:95:5)
    at /home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:284:15
    at Function.process_params (/home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:346:12)
    at next (/home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:280:10)
    at /home/dnm/nodejs-poolController/web/Server.ts:749:25
    at Layer.handle [as handle_request] (/home/dnm/nodejs-poolController/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:328:13)
    at /home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:346:12)
    at next (/home/dnm/nodejs-poolController/node_modules/express/lib/router/index.js:280:10)
    at /home/dnm/nodejs-poolController/node_modules/body-parser/lib/read.js:137:5
> 

![Screenshot 2023-05-13 6 42 03 AM](https://github.com/tagyoureit/nodejs-poolController/assets/34433189/f21eadbb-4f39-49d3-93e9-c1f2950971df)

**Comment by rstrouse** [comment]
Ok so I found the replay that was sandwiched in between the two screenshots.  Sucker was hiding from me.  From this I determined that ScreenLogic asks for the configuration from the old IntelliFlo4 and in a ridiculous twist EasyTouch returns garbage.  We were seeing this garbage and picked it up as gospel without first checking the firmware version.  We assumed it would never show up if it were not correct.

Anyway pull njsPC then please do a replay that includes reloading the config.

**Comment by dcorrea78** [comment]
I think you are asking me for an updated replay. I can do that in a few hours when I get home. Forgive my ignorance, but when you say pull njsPC, are you saying to do a git pull to update njsPC?

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Reply by rstrouse** [reply]
Yes git pull to update njsPC.

**Comment by johnny2678** [comment]
I'll attach my results. 

I *think* I'm good now.

Pump config page is showing correct rpm circuits:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/d3337995-341b-402c-b5ac-a343c1b186b1)

Not sure I took the right route to git here but I'll post in case you have any comments.

Did a git pull/npm i. Then started a replace with config reloaded.

The config came up completely empty, as if I had a new nixie install with nothing configured, instead of my previously configured ET4 config:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/e9b6fbb0-e169-401c-a03c-5d22fa7128f3)

replay of the reload config here:
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11470555/replay.zip)

I didn't like the look of that so I started a second replay and did a full stack pm2 restart.

This time, my ET4 config reloaded and my pump config page looked good.  Just had to hide an AUX circuit that shouldn't have been showing and I was back in business.

2nd replay here: 
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/11470565/replay.1.zip)

bottom line, I think I'm good?  Unless you see something in the replay you don't like.

Thanks for looking at this.  Hope this works for @dcorrea78 as well.

**Reply by rstrouse** [reply]
Set your body capacities under the configuration.  It is important.

**Reply by johnny2678** [reply]
They were set. Think it just took a service restart to get them to show

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/189afebd-990d-4367-8742-5bfcea2218d5)

**Comment by dcorrea78** [comment]
Here's my replay. I also had to do a pm2 restart to get the config to reload properly. 
[replay (5).zip](https://github.com/tagyoureit/nodejs-poolController/files/11470858/replay.5.zip)

**Comment by dcorrea78** [comment]
I went all day today with no issues. Looks like everything is fixed. Thank you!
